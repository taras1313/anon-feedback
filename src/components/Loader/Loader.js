import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';

const Loader = ({ height, width }) => {
	const useStyles = makeStyles(theme => ({
		root: {
			width: width || '100%',
			height: height || '100%',
			display: 'flex',
			justifyContent: 'center',
			alignItems: 'center',
			'& > * + *': {
				margin: 'auto'
			}
		}
	}));

	const classes = useStyles();

	return (
		<div className={classes.root}>
			<CircularProgress />
		</div>
	);
};

export default Loader;
