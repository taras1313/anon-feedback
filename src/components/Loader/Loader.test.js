import React from 'react';
import Loader from './Loader';
import { shallow } from 'enzyme';

describe('Loader component', () => {
	test('render', () => {
		const wrapper = shallow(<Loader />);

		expect(wrapper).toMatchSnapshot();
	});
});
