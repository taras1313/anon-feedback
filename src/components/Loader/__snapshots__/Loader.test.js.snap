// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component render 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(CircularProgress)) />
</div>
`;
